packages/rum/src/userActionCollection.ts

```diff
@@ -1,24 +1,50 @@
-import { Context, DOM_EVENT, generateUUID, monitor, Observable } from '@datadog/browser-core'
+import { Context, DOM_EVENT, generateUUID } from '@datadog/browser-core'
 import { getElementContent } from './getElementContent'
 import { LifeCycle, LifeCycleEventType, Subscription } from './lifeCycle'
 import { trackEventCounts } from './trackEventCounts'
+import { waitIdlePageActivity } from './trackPageActivities'
+import { View } from './viewCollection'
 
 // Automatic user action collection lifecycle overview:
-//
-//                               (Start)
-//                   .--------------'----------------------.
-//                   v                                     v
+//                           (Start new view)
+//                 .------------------'------------------.
+//                 v                                     v
+//     [Wait for a page activity ]          [Wait for a maximum duration]
+//     [timeout: VALIDATION_DELAY]          [  timeout: MAX_DURATION    ]
+//         /                  \                             |
+//        v                    v                            |
+// [No page activity]   [Page activity]                     |
+//        |                    |,-----------------------.   |
+//        |                    v                        |   |
+//        |          [Wait for a page activity]         |   |
+//        |          [   timeout: END_DELAY   ]         |   |
+//        |              /                \             |   |
+//        |             v                  v            |   |
+//        |     [No page activity]    [Page activity]   |   |
+//        |             |                  |            |   |
+//        |             |                  '------------'   |
+//        |             |                                   |
+//        '-------------'----------. ,----------------------'
+//                                  v
+//                         (View load complete)
+//                                  |
+//                                  |
+//                                  |
+//                                  v
+//                        (Start new user action)
```

Comment:
This schema can be moved to the "trackPageActivities" module (also, since it is the same for user action and view load duration, no need to put it twice)

Reply:
In one case the [No page activity]  leads to discard the UA, in the other, it leads to the end of the view load so they are not exactly similar. But I get you point and moving it to the trackPageActivities makes sense